lib/runtime/procs/read.js

```diff
@@ -9,10 +9,13 @@ var JuttleMoment = require('../../runtime/types/juttle-moment');
 var Promise = require('bluebird');
 
 
-var Read = source.extend({
-    procName: 'read',
+class Read extends source {
+    get procName() {
```

Comment:
This is a bit tricky since we need the `procName` to be set in the base constructor I actually think it would be a better change if the procName were passed into the proc from the compiler and set in the base proc constructor so we don't need to have all this repetition everywhere.

I'll give this a shot and we can see what we think.